src/v3/components/search_bar/index.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { StyledBox, StyledInput, StyledButton } from './search_bar.styled';
+import { IconSearch, IconClose } from '../icons';
+import { InputAdornment } from '@mui/material';
+import { SearchBarProps } from './search_bar.types';
+
+const SearchBar = ({ placeholder, onSearch }: SearchBarProps) => {
+  const [value, setValue] = useState('');
+
+  const handleChange = (event) => {
+    const query = event.target.value;
+    setValue(query);
+    handleSearch(query);
+  };
+
+  const handleKeyDown = (event) => {
+    if (event.key === 'Enter') {
+      handleSearch(value);
+      event.target.blur();
+    }
+  };
+
+  const handelSearchClick = (event) => {
+    handleSearch(value);
+    event.target.blur();
+  };
+
+  const handleSearch = (query) => {
+    const searchQuery = query.trim();
+
+    if (searchQuery.length > 0) {
+      onSearch(searchQuery);
+    } else {
+      onSearch('');
+    }
+  };
+
+  const handleClear = () => {
+    setValue('');
+    onSearch('');
+  };
+
+  return (
+    <>
```

Comment:
@IlkaA1: could please fix the code here as reported by sonarcloud [here](https://sonarcloud.io/project/issues?resolved=false&pullRequest=997&id=sws2apps_cpe-sws&open=AY2N_vN78YUq_STiMjc1)? Thank you.